scripts/esbuild-vue-plugin.js

```diff
@@ -0,0 +1,108 @@
+// Modified from https://github.com/apeschar/esbuild-vue to remove the Piscina dependency
+// and add path aliasing support in at-import rules.
+'use strict'
+
+const { readFile } = require('fs').promises
+const { relative } = require('path')
+
+const componentCompiler = require('@vue/component-compiler')
+
+const { createAliasReplacer } = require('./utils')
+
+module.exports = ({ aliases, debug = false } = {}) => {
+  const aliasReplacer = aliases ? createAliasReplacer(aliases) : null
+
+  return {
+    name: 'vue',
+    setup (build) {
+      build.onLoad({ filter: /[^/]\.vue$/ }, async ({ path }) => {
+        const filename = relative(process.cwd(), path)
+        const source = await readFile(path, 'utf8')
+          .then(source => aliasReplacer ? aliasReplacer({ path, source }) : source)
+
+        if (debug) console.log('Compiling:', filename)
+        const { result /*, usedFiles */ } = await compile({ filename, source })
+
+        return result
+      })
+    }
+  }
+}
+
+let requireDepth = 0
+let usedFiles = new Set()
+
+editModule('fs', (fs) => {
+  fs.readFileSync = new Proxy(fs.readFileSync, {
+    apply (target, thisArg, args) {
+      if (usedFiles && requireDepth === 0) usedFiles.add(args[0])
+      return Reflect.apply(target, thisArg, args)
```

Comment:
Interesting, I've never seen `Reflect.apply` before even though I've been coding JS for many years. Crazy!

I'm curious, what is the reason you chose to do this instead of `target.apply`?

Oh - this is probably not your code, right? I have no problem with this btw, just curious.

Reply:
Yea that code comes from the original plugin, namely from https://github.com/apeschar/esbuild-vue/blob/master/src/worker.js